### DISCUSSION TITLE
How to Add Folders to Final Build

### DISCUSSION THREAD

**Discussion by VitorAndrey** [alex8088/electron-vite#435] [Q&A]
Hello Electron + Vite community!👋,

I'm encountering an issue when adding folders to the final build of my project using Electron + Vite. During development, everything works perfectly; however, after building the application, an error is thrown, indicating that the migrations folder cannot be found, as if the path used in development is not valid after the build.

Here is where I use the migrations folder path, which works in development, but after the build, it seems that ```src/database/migrations``` does not resolve to the right path.

```ts
import { migrate } from 'drizzle-orm/better-sqlite3/migrator'

migrate(db, { migrationsFolder: 'src/database/migrations' })
```

![image](https://github.com/alex8088/electron-vite/assets/114686823/917942bb-d7dc-4c40-a4f3-6b26f5872d5b)

The problem was previously discussed in the context of Electron + Webpack on [Discord](https://discordapp.com/channels/1043890932593987624/1205153107231772683), where a user resolved the problem by manually copying the necessary folder to the output package. The solution using Webpack involved adding configurations to manually copy the migrations folder during the build.

```ts
const webpackConfig = {
  // ...rest of webpack config
  plugins: [
    // ... other plugins 
    new CopyWebpackPlugin({
      patterns: [
        {
          from: path.join(webpackPaths.srcMainPath, 'db/drizzle'),
          to: path.join(webpackPaths.distMainPath, 'db/drizzle'),
        },
      ],
    }),
  ],
}
```

Now, I am seeking guidance on how to achieve the same result in the context of Electron + Vite. If anyone can provide insights or specific suggestions on configuring Electron + Vite to include necessary folders in the final build, I would greatly appreciate it.

The error that occurs after the build suggests that the /database folder may not be included in the built version.

![Folder](https://github.com/alex8088/electron-vite/assets/114686823/42dcc695-5ca5-43ed-a64b-fd761939eaaa)

Thank you for your attention and support!

**Comment by alex8088** [comment]
check out https://electron-vite.org/guide/assets

**Comment by alex8088** [comment]
```js
import jsonFile from '../../resources/xx.json?commonjs-external&asset'

// get the public dir path
const dir = path.dirname(jsonFile)
```

**Reply by VitorAndrey** [reply]
@alex8088  So should I use the `dir` path instead of `src/database/migrations`?

For example:

```ts
import { migrate } from 'drizzle-orm/better-sqlite3/migrator';
import jsonFile from '../../resources/xx.json?commonjs-external&asset';

const dir = path.dirname(jsonFile);
migrate(db, { migrationsFolder: dir });
```

**Reply by alex8088** [reply]
Yes, use this way to find your migrations dir.

**Reply by VitorAndrey** [reply]
@alex8088, Thanks so much for spending time to help. You have been very attentive, but I have one more question. Do I have to move my actual migrations folder into the resources folder? I tried making the change above, but I faced this error while building with the `build:win` command.

````src/database/index.ts``` with sugested  changes:
```ts
// Other imports
import jsonFile from '../../resources/xx.json?commonjs-external&asset'

// Irrelevant rest of code creating "db" used bellow

const dir = path.dirname(jsonFile)
migrate(db, { migrationsFolder: dir })
```

Build command crash error:
```bash
vite v5.0.12 building SSR bundle for production...
✓ 9 modules transformed.
Could not resolve "../../resources/xx.json?commonjs-external&asset" from "src/database/index.ts"
file: C:/Users/vitor/projects/somonitores/transmissor-de-arquivos/src/database/index.ts
error during build:
RollupError: Could not resolve "../../resources/xx.json?commonjs-external&asset" from "src/database/index.ts"
    at error (file:///C:/Users/vitor/projects/somonitores/transmissor-de-arquivos/node_modules/rollup/dist/es/shared/parseAst.js:337:30)
    at ModuleLoader.handleInvalidResolvedId (file:///C:/Users/vitor/projects/somonitores/transmissor-de-arquivos/node_modules/rollup/dist/es/shared/node-entry.js:18022:24)
    at file:///C:/Users/vitor/projects/somonitores/transmissor-de-arquivos/node_modules/rollup/dist/es/shared/node-entry.js:17982:26
```

![move-migrations](https://github.com/alex8088/electron-vite/assets/114686823/d02723a5-4057-4d8c-ab06-3d6b34fc484c)

**Reply by alex8088** [reply]
You should import a real existing file.
